core/arch/asm_defines.asm

```diff
@@ -910,11 +910,12 @@ ASSUME fs:_DATA @N@\
         mov      ARG1, p1   @N@\
         blx      callee
 #elif defined(RISCV64)
-/* For RISC-V, there is not a instruction could operate both immediate
- * and registers. Here is a macro which to judge if the arguments is
- * registers or not.
+/* For RISC-V, there is no a instruction can operate on both immediates
+ * and registers. Here is a macro which judges whether its argument is
+ * a register or no.
  */
 .set register.sp, 1
```

Comment:
There is not much work to do to complete this implementation. No need to add a TODO IMO. Also, I'd like to use `mov` instead of `RVSETARG`, which is more consistent with X86/ARM port, so something like this:

```
/* HACK: Mimics an ARM-like MOV instruction. */
.set reg.sp     1
.set reg.x0     1
.set reg.x1     1
.set reg.x2     1
.set reg.x3     1
.set reg.x4     1
.set reg.x5     1
.set reg.x6     1
.set reg.x7     1
.set reg.x8     1
.set reg.x9     1
.set reg.x10    1
.set reg.x11    1
.set reg.x12    1
.set reg.x13    1
.set reg.x14    1
.set reg.x15    1
.set reg.x16    1
.set reg.x17    1
.set reg.x18    1
.set reg.x19    1
.set reg.x20    1
.set reg.x21    1
.set reg.x22    1
.set reg.x23    1
.set reg.x24    1
.set reg.x25    1
.set reg.x26    1
.set reg.x27    1
.set reg.x28    1
.set reg.x29    1
.set reg.x30    1
.set reg.x31    1
.macro mov reg, p
  .ifdef "reg.\p"
        mv      \reg, \p
  .else
        li      \reg, \p
  .endif
.endm
```